yt_dlp/extractor/twitch.py

```diff
@@ -306,6 +348,26 @@ def _extract_info(info):
 
     @staticmethod
     def _extract_info_gql(info, item_id):
+        def moment_to_chapter(moment):
+            moment = moment.get('node')
+            if moment.get('type') != 'GAME_CHANGE':
+                # TODO: warning or debug log if found unknown moment type
+                return None  # TODO: or empty dict?
+
+            momentPosition = int_or_none(moment.get('positionMilliseconds'), 1000)
+            momentDuration = int_or_none(moment.get('durationMilliseconds'), 1000)
+
+            if momentPosition is None or momentDuration is None:
+                # TODO: warning or debug log about missing important data
+                chapters.clear()
```

Comment:
1. `chapters` should be `None` (not `[]`) if chapter data extraction fails
2. It is bad practice to clear a nonlocal variable like this, especially in the middle of assigning it